### PR TITLE
[Improve] Update apis and packages.

### PR DESCRIPTION
**Pull request by mzr1996**
**This is a PR for mmcls 1.x**

## Modification

1. Update `apis/inference.py` and demo to make the demo available.
2. Remove the entire `mmcls/core` package. `mmcls.core.EvalHook` is already deprecated, and `mmcls.core.WandbLoggerHook` will be re-implemented later.
3. Update the mmcv dependency version.

### CODE REVIEW

File: mmcls/apis/inference.py [hunk 1]
```diff
@@ -55,66 +56,34 @@ def inference_model(model, img):
...
+    data = pseudo_collate([data])
 
     # forward the model
     with torch.no_grad():
-        scores = model(return_loss=False, **data)
-        pred_score = np.max(scores, axis=1)[0]
-        pred_label = np.argmax(scores, axis=1)[0]
+        prediction = model.val_step(data)[0].pred_label
+        pred_score = torch.max(prediction.score).item()
+        pred_label = torch.argmax(prediction.score).item()
```
Comment (Ezra-Yu) [inline mmcls/apis/inference.py:85]: This does work in multi-label classification tasks, why not just return a pred label?

File: mmcls/apis/inference.py [hunk 2]
```diff
@@ -55,66 +56,34 @@ def inference_model(model, img):
     """Inference image(s) with the classifier.
 
     Args:
-        model (nn.Module): The loaded classifier.
+        model (BaseClassifier): The loaded classifier.
```
Comment (yingfhu) [inline mmcls/apis/inference.py:59]: `mmcls.BaseClassifier` should be clearer

File: mmcls/apis/inference.py (hunk 1 above)
Comment (mzr1996) [reply mmcls/apis/inference.py:85]: This is just a temporary fix to ensure the original result.

File: mmcls/apis/inference.py [hunk 3]
```diff
@@ -55,66 +56,34 @@ def inference_model(model, img):
...
         img (str/ndarray): The image filename or loaded image.
```
Comment (yingfhu) [inline mmcls/apis/inference.py:60]: str | np.ndarray

**Review by Ezra-Yu — APPROVED** [review]
LGTM.

File: mmcls/__init__.py [hunk 4]
```diff
@@ -5,8 +5,8 @@
 
 from .version import __version__
 
-mmcv_minimum_version = '1.4.2'
-mmcv_maximum_version = '1.6.0'
+mmcv_minimum_version = '2.0.0rc0'
+mmcv_maximum_version = '2.0.0'
```
Comment (yingfhu) [inline mmcls/__init__.py:9]: is extra version buffer needed here? e.g. 2.2.0

File: mmcls/apis/inference.py (hunk 3 above)
Comment (mzr1996) [reply mmcls/apis/inference.py:60]: Fixed

File: mmcls/__init__.py (hunk 4 above)
Comment (mzr1996) [reply mmcls/__init__.py:9]: `rc` will be a long-term tag, it includes all `rc0`, `rc1`, and all.
